internal/controller/api/v1alpha1/installenv_types.go

```diff
@@ -21,6 +21,49 @@ import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
+const (
+	ImageStateCreated        = "Image has been created"
+	ImageStateFailedToCreate = "Failed to create image"
+)
+
+// ClusterReference represents a Cluster Reference. It has enough information to retrieve cluster
+// in any namespace
+type ClusterReference struct {
```

Comment:
because the manifests generator looks for a JSON tag, which it doesn't have:

```golang
+ controller-gen crd:trivialVersions=true rbac:roleName=manager-role paths=./... output:rbac:dir=/home/nmagnezi/go/src/github.com/assisted-service/internal/controller/config/rbac webhook paths=./... output:crd:artifacts:config=/home/nmagnezi/go/src/github.com/assisted-service/internal/controller/config/crd/bases
/go/pkg/mod/k8s.io/apimachinery@v0.19.5/pkg/types/namespacedname.go:32:2: encountered struct field "Namespace" without JSON tag in type "NamespacedName"
/go/pkg/mod/k8s.io/apimachinery@v0.19.5/pkg/types/namespacedname.go:33:2: encountered struct field "Name" without JSON tag in type "NamespacedName"
Error: not all generators ran successfully
```

Reply:
Yup. the more that I think about this the less sense it makes to try and use NamespacedName.